src/extension.ts

```diff
@@ -38,27 +39,74 @@ export function activate(context: vscode.ExtensionContext) {
   // listen for configuration changes
   context.subscriptions.push(
     vscode.workspace.onDidChangeConfiguration((e) => {
-      const configChanged = e.affectsConfiguration('vimFindHighlight.charPrimaryColor') || e.affectsConfiguration('vimFindHighlight.charSecondaryColor') || e.affectsConfiguration('vimFindHighlight.charFontWeight') || e.affectsConfiguration('vimFindHighlight.enableUnderline');
+      const configChanged =
+        e.affectsConfiguration("vimFindHighlight.charPrimaryColor") ||
+        e.affectsConfiguration("vimFindHighlight.charSecondaryColor") ||
+        e.affectsConfiguration("vimFindHighlight.charFontWeight") ||
+        e.affectsConfiguration("vimFindHighlight.enableUnderline") ||
+        e.affectsConfiguration("vimFindHighlight.enableAutoHighlight");
 
       if (configChanged) {
         configureDecoration();
+        autoHighlight = configureAutoHighlight();
       }
     })
-  )
+  );
 
-  // when activating the extension. read the user settings file first
+  // when activating the extension. read the user settings file first and register the commands
   configureDecoration();
+  registerCommands(context);
 }
 
 const configureDecoration = () => {
   updateDecorationConfig();
   disposeCharDecoration();
 };
 
+const configureAutoHighlight = (): boolean => {
+  const settings = vscode.workspace.getConfiguration();
+  const autoHighlight = settings.get(
+    "vimFindHighlight.enableAutoHighlight"
+  ) as boolean;
+  vscode.window.showInformationMessage(autoHighlight.toString());
+  return autoHighlight;
+};
+
 const main = (cursorPos: number, currentLine: string) => {
   const toColor = charHighlighter.getCharHighlighting(currentLine, cursorPos);
   colorChars(toColor, decorationConfig);
 };
 
+const registerCommands = (context: vscode.ExtensionContext) => {
+  const toggleCommand = "vimFindHighlight.toggleAuthoHighlight";
+  context.subscriptions.push(
+    vscode.commands.registerCommand(toggleCommand, toggleAutoHighlight)
+  );
+
+  const highlightCommand = "vimFindHighlight.highlightCharacters";
+  context.subscriptions.push(
+    vscode.commands.registerCommand(highlightCommand, highlightCharacters)
+  );
+};
+
+const highlightCharacters = () => {
+  const line = getCurrentLine();
+  const cursorPos = getCursorPos();
+  if (line?.text.length && cursorPos) {
+    main(cursorPos, line.text);
+  } else {
+    disposeCharDecoration();
+  }
+};
+
 // this method is called when your extension is deactivated
-export function deactivate() { }
+export function deactivate() {}
+
+export function toggleAutoHighlight() {
+  const settings = vscode.workspace.getConfiguration();
+  const autoHighlight = settings.get(
+    "vimFindHighlight.enableAutoHighlight"
+  ) as boolean;
+  settings.update("vimFindHighlight.enableAutoHighlight", !autoHighlight);
```

Comment:
During testing, nothing changed for me after toggling the setting. This was because this update command failed. Currently, we are not showing the user that it failed. It failed for me since i didn't debug in a workspace. Once i debugged in a workspace, it worked perfectly and the toggle caused the chars not to be highlighted.

```suggestion
    settings.update("vimFindHighlight.enableAutoHighlight", !autoHighlight).
    then(
      () => { },
      (reason) => vscode.window.showInformationMessage("Failed to toggle auto highlight: " + reason),
    );
```

It will look like this:
<img width="1440" alt="Screenshot 2024-08-31 at 11 38 31" src="https://github.com/user-attachments/assets/3bd9bff4-a628-4a19-8714-68aab6cb4638">

Reply:
Mmmm interesting that it only works in a workspace. Do you know if that's common among other extensions? Should we update user setting instead? I have no idea what's common practice or expected behavior for vscode to be honest!

Reply:
Yes i think we need to change the value in the global settings since users will probably either want it everywhere or not want it everywhere as well. There seems to be an answer here https://stackoverflow.com/questions/68526627/vscode-extension-how-do-i-edit-workspace-global-config.

Reply:
Cool! Will check it out this afternoon!

Reply:
I added `vscode.ConfigurationTarget.Global` and did the error handling suggested in the other comment (just in case).